source/MRMesh/MRCylinder3.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+#include "MRLine.h"
+
+namespace MR
+{
+// A class describing a cylinder as a mathematical object.A cylinder is represented by a centerline, a radius, and a length.template <typename T>
+// TODO: Cylinder3 could be infinite. For example for infinite Cylinder3 we could use negative length or length = -1
+template <typename T>
+class Cylinder3
+{
+public:
+    Cylinder3()
+    {}
+
+    Cylinder3( const Vector3<T>& inCenter, const Vector3<T>& inDirectoin, T inRadius, T inHeight )
+        :
+        mainAxis( inCenter, inDirectoin ),
+        radius( inRadius ),
+        length( inHeight )
+    {}
+    Cylinder3( const Line3<T>& inAxis, T inRadius, T inHeight )
+        :
+        mainAxis( inAxis ),
+        radius( inRadius ),
+        length( inHeight )
+    {}
+
+    inline MR::Vector3<T>& center( void )
+    {
+        return mainAxis.p;
+    }
+    inline MR::Vector3<T>& direction( void )
+    {
+        return mainAxis.d;
+    }
+
+    MR::Line3<T> mainAxis = {};
```

Comment:
Without `= {}` it will be zero initialized anyway.

Reply:
done